### PR TITLE
12-voting-database-model

### LINKED ISSUE #12: Voting
#67  Add voting capabilities to transcripts

- Database tables to hold voting info
- User should be able to vote (should be able to hit "like" and"unlike") 
- Users need to be authenticated and can only vote once per transcript.

### PR DESCRIPTION
**Pull request by leilafarsani**
✨ **New feature**:  Added voting capabilities for transcriptions.
🐛 Bug fix: N/A 
♻️ Refactor: N/A 
⚙️ Chore: N/A 

## Description 

✨ Added voting capabilities: Implemented a new Vote model for rating transcriptions.
🔄 Modified Transcription model: Added a votes field to track multiple votes for a transcription.
✅ Ensured unique votes: Added a constraint to allow users to vote only once for each transcription.

### CODE REVIEW

**Comment by susanssky** [comment]
The post already has votes, no need to open a new model. Or can you explain in detail?

**Comment by leilafarsani** [comment]
Thanks for sharing your thoughts. I understand your point, but I still think it's best to have a separate model for voting. This approach helps us handle the specific aspects of voting, like user authentication, one vote per transcript, and a 1-5 rating scale. It also keeps our code organized, maintainable, and scalable. Let me know what do you think?

**Comment by susanssky** [comment]
I won't approve until I complete `create posts`, please be patient

File: prisma/schema.prisma [hunk 1]
```diff
@@ -23,16 +23,17 @@ model Sentence {
...
+  sentences        Sentence[]
+  createdAt        DateTime     @default(now())
+  updatedAt        DateTime     @updatedAt
+  post             Post?
+  postId           Int          @unique
+  title            String?
+  keywords         String?
+  categories       String?
+  speaker          String?
+  votes            Vote[]
```
Comment (maxf) [inline prisma/schema.prisma]: This shouldn't be needed, as the `Vote` model contains all the information.
Gathering all the votes for a video should be done in SQL, like 
```
SELECT * FROM Vote WHERE Vote.transcription = transcription.id
```

File: prisma/schema.prisma [hunk 2]
```diff
@@ -55,19 +56,21 @@ model Account {
...
+  email         String?   @unique
+  emailVerified DateTime?
+  image         String?
+  role          Role?     @default(Trainee)
+  accounts      Account[]
+  createdAt     DateTime  @default(now())
+  updatedAt     DateTime  @updatedAt
+  posts         Post[]
+  postId        Int?
+  votes         Vote[]
```
Comment (maxf) [inline prisma/schema.prisma:70]: as above

File: prisma/schema.prisma [hunk 3]
```diff
@@ -89,3 +92,16 @@ model Post {
     createdAt          DateTime      @default(now())
     updatedAt          DateTime      @updatedAt
 }
+
+model Vote {
+  id               Int          @id @default(autoincrement())
+  user             User         @relation(fields: [userId], references: [id])
+  userId           String
+  transcription    Transcription @relation(fields: [transcriptionId], references: [id])
```
Comment (maxf) [inline prisma/schema.prisma]: I don't know prisma, sorry. What does this mean, and how does it translate to SQL?

**Comment by maxf** [comment]
Try to avoid committing `.DS_Store` files by adding the name in your `.gitignore` file

**Comment by maxf** [comment]
By the way, your PR has too many commits for the changes you made. And each commit doesn't have a very useful message. Talk to me if you need more advice

**Comment by susanssky** [comment]
> Thanks for sharing your thoughts. I understand your point, but I still think it's best to have a separate model for voting. This approach helps us handle the specific aspects of voting, like user authentication, one vote per transcript, and a 1-5 rating scale. It also keeps our code organized, maintainable, and scalable. Let me know what do you think?

I want all the functions to be centralised in the post model, including the voting function, thank you

File: prisma/schema.prisma (hunk 3 above)
Comment (susanssky) [reply prisma/schema.prisma]: Prisma is a modern ORM (Object-Relational Mapping) framework that can assist developers in managing and operating databases more conveniently. When using Prisma, it automatically converts your Prisma model file into the corresponding SQL commands.

Once you have defined a Prisma model, Prisma will automatically generate the appropriate SQL commands for you.

Comment (maxf) [reply prisma/schema.prisma]: Thanks. So what's the difference between `transcription` and `transcriptionId` in this model?

Comment (susanssky) [reply prisma/schema.prisma]: In the above data model, "Transcription" and "transcriptionId" are associated with each other. "Transcription" is a relationship field that points to the "Transcription" object in another model, while "transcriptionId" is a foreign key that stores the unique identifier of the "Transcription" object associated with the "Sentence" object.

Simply put, "Transcription" is a relationship field that allows access to the associated "Transcription" object and its properties. "transcriptionId", on the other hand, is a regular field that simply stores the unique identifier of the associated object. Typically, ORM frameworks use "transcriptionId" for querying and association operations.

**Comment by susanssky** [comment]
@leilafarsani Now I created the PR for `create posts`, so your PR (I think) need update from my PR, thank you

**Comment by susanssky** [comment]
> @leilafarsani Now I created the PR for `create posts`, so your PR (I think) need update from my PR, thank you

ok now??

**Pull request closed without merging**